core/src/main/java/org/apache/calcite/plan/SubstitutionVisitor.java

```diff
@@ -1130,20 +1130,23 @@ protected MutableRel invert(MutableRel model, MutableRel input,
       }
       final List<RexNode> exprList = new ArrayList<>();
       final RexBuilder rexBuilder = model.cluster.getRexBuilder();
-      for (RelDataTypeField field : model.rowType.getFieldList()) {
-        exprList.add(rexBuilder.makeZeroLiteral(field.getType()));
+      for (int i = 0; i < model.rowType.getFieldCount(); i++) {
+        exprList.add(null);
```

Comment:
Worth adding a comment to explain the intent of doing this.

Reply:
Sure, will do :)